server/chat/backend/agent/tools/memory_tool.py

```
@@ -0,0 +1,557 @@
+                    )
+                else:
+                    cursor.execute(
+                        """SELECT title, category, content FROM artifacts
+                           WHERE org_id = %s AND category = ANY(%s)
+                           AND content ~* %s
+                           ORDER BY updated_at DESC
+                           LIMIT %s""",
+                        (org_id, list(MEMORY_CATEGORIES), search_pattern, _GREP_MAX_RESULTS),
+                    )
+                rows = cursor.fetchall()
+            except Exception as regex_err:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)